### PR TITLE
Add persistent volume claim retention policy

### PR DESCRIPTION
**Pull request by romsharon98**
closes: https://github.com/apache/airflow/pull/38361
Redo stale PR. Commits where cherry picked from stale PR to preserve credit to the origial author
cc: @nirroz93

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: chart/values.schema.json [hunk 1]
```diff
@@ -1591,6 +1591,27 @@
...
+                                    "type": "string",
+                                    "default": "Retain"
+                                },
+                                "whenScaled": {
+                                    "description": "Whether to retain the PVC when the StatefulSet is scaled.",
+                                    "type": "string",
+                                    "default": "Retain"
+                                }
+                            }
+                        },
```
Comment (hussein-awala) [inline chart/values.schema.json]: You can create a new definition in `definitions` to reuse instead of duplicating it twice.

**Review by hussein-awala — APPROVED** [review]
I added a small nit, otherwise LGTM

**Review by eladkal — APPROVED** [review]

File: chart/values.schema.json [hunk 2]
```diff
@@ -1591,6 +1591,27 @@
                             "type": "boolean",
                             "default": true
                         },
+                        "persistentVolumeClaimRetentionPolicy": {
+                            "description": "How PVCs are deleted during the lifecycle of a StatefulSet",
```
Comment (amoghrajesh) [inline chart/values.schema.json]:
```suggestion
                            "description": "PersistentVolumeClaim retention policy to be used in the lifecycle of a StatefulSet",
```

File: chart/values.schema.json [hunk 3]
```diff
@@ -2936,6 +2957,26 @@
                             "type": "string",
                             "default": "100Gi"
                         },
+                        "persistentVolumeClaimRetentionPolicy": {
+                            "description": "How PVCs are deleted during the lifecycle of a StatefulSet",
```
Comment (amoghrajesh) [inline chart/values.schema.json]:
```suggestion
                            "description": "PersistentVolumeClaim retention policy to be used in the lifecycle of a StatefulSet",
```

File: chart/values.yaml [hunk 4]
```diff
@@ -619,6 +619,11 @@ workers:
   persistence:
     # Enable persistent volumes
     enabled: true
+    # persistentVolumeClaimRetentionPolicy if statefulset is used, to determine delete when scaled/removed
```
Comment (amoghrajesh) [inline chart/values.yaml]: Can we add a better comment here? This is not too accurate / is confusing

File: chart/values.yaml [hunk 5]
```diff
@@ -1532,6 +1537,8 @@ triggerer:
...
```
Comment (amoghrajesh) [inline chart/values.yaml]: Same comment as above

**Review by amoghrajesh — APPROVED** [review]
I have a few nits, otherwise looks good @romsharon98

File: chart/values.schema.json (hunk 1 above)
Comment (romsharon98) [reply chart/values.schema.json]: move to `definitions`

File: chart/values.schema.json (hunk 2 above)
Comment (romsharon98) [reply chart/values.schema.json]: thanks, changed to your suggestion

File: chart/values.yaml (hunk 4 above)
Comment (romsharon98) [reply chart/values.yaml]: changed, let me know what you this now

**Pull request merged**